app/src/main/kotlin/no/nav/arbeidsgiver/notifikasjon/ekstern_varsling/EksternVarslingStatusEksportService.kt

```diff
@@ -0,0 +1,111 @@
+import java.util.*
+
+class EksternVarslingStatusEksportService(
+    val eventSource: HendelsesstrømKafkaImpl,
+    val repo: EksternVarslingRepository,
+    val kafka: KafkaProducer<String, VarslingStatusDto> = createKafkaProducer(),
+) {
+
+    fun start(coroutineScope: CoroutineScope): Job {
+        return coroutineScope.launch {
+            eventSource.forEach { event, meta ->
+                prosesserHendelse(event, meta)
+            }
+        }
+    }
+
+    suspend fun prosesserHendelse(
+        event: HendelseModel.Hendelse,
+        meta: HendelseModel.HendelseMetadata
+    ) {
+        val varslingStatus = when (event) {
+            is HendelseModel.EksterntVarselFeilet -> {
+                val status = when (event.altinnFeilkode) {
+                    "30304",
+                    "30307",
+                    "30308" -> Status.MANGLER_KOFUVI
+
+                    else -> Status.ANNEN_FEIL
+                }
+
+                lagVarslingStatus(
+                    status = status,
+                    virksomhetsnummer = event.virksomhetsnummer,
+                    varselId = event.varselId,
+                    hendelseTimestamp = meta.timestamp.asOsloLocalDateTime()
+                ) ?: return
+            }
+
+            is HendelseModel.EksterntVarselVellykket -> {
+                lagVarslingStatus(
+                    status = Status.OK,
+                    virksomhetsnummer = event.virksomhetsnummer,
+                    varselId = event.varselId,
+                    hendelseTimestamp = meta.timestamp.asOsloLocalDateTime()
+                ) ?: return
+            }
+
+            else -> return
+        }
+
+        kafka.suspendingSend(ProducerRecord(TOPIC, varslingStatus))
+    }
+
+    private suspend fun lagVarslingStatus(
+        status: Status,
+        virksomhetsnummer: String,
+        varselId: UUID,
+        hendelseTimestamp: LocalDateTime
+    ) = repo.findVarsel(varselId)?.let { varsel ->
+        VarslingStatusDto(
+            virksomhetsnummer = virksomhetsnummer,
+            varselId = varselId,
+            varselTimestamp = varsel.data.eksternVarsel.sendeTidspunkt
+                ?: varsel.kalkuertSendetidspunkt(hendelseTimestamp),
+            eventTimestamp = hendelseTimestamp,
+            status = status
+        )
+    }
+}
+
+
+const val TOPIC = "fager.ekstern-varsling-status"
+
+private fun createKafkaProducer() = KafkaProducer<String, VarslingStatusDto>(
+    COMMON_PROPERTIES + SSL_PROPERTIES + mapOf<String, Any>(
+        ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to StringSerializer::class.java.canonicalName,
+        ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to VarslingStatusDtoSerializer::class.java.canonicalName,
+        ProducerConfig.MAX_BLOCK_MS_CONFIG to 5_000,
+        ProducerConfig.ACKS_CONFIG to "all",
+    )
+)
+
+data class VarslingStatusDto(
+    val virksomhetsnummer: String,
+    val varselId: UUID,
+    val varselTimestamp: LocalDateTime,
```

Comment:
Sitter og funderer på disse to `LocalDateTime`-ene. Begge er timestamps (i betydningen tidspunkter for når noe har skjedd), så på en måte virker det riktigere å bruke UTC (`Instant`). Samtidig, så må det jo uansett konverteres tilbake til _Europe/Oslo_ for å vises til brukeren ...

Reply:
kilden til varselTimestamp er en LocalDateTime, så den tenker jeg er grei.
kilden til eventTimestamp er en Instant, så det kan vi godt beholde som instant. Her gikk jeg for samme type på begge da jeg tenkte det ikke var så viktig. eventTimestamp brukes kun til å sørge for at vi ser på siste status i andre enden

Reply:
158039f7a78061b37364898f2b68d300cc6671b0